.github/workflows/ci.yaml

```
@@ -35,3 +43,44 @@ jobs:
 
     - name: Unit Tests
       run: php ./vendor/bin/pest
+  docker:
+    if: ${{ needs.build-test.result == 'success' && github.event_name != 'pull_request'}}
+# env is not avaialble in context of evaluating condition or building another env. workaround was to use seperate steps to construct VERSION_TAG env variable
+    # env:
+        # VERSION_TAG: ${{ startsWith(github.ref, 'refs/tags/v') && env.CRATER_VERSION || env.CRATER_VERSION-github.run_number }}
+    name: Docker build and publish
+    runs-on: ubuntu-latest
+    needs:
+      - build-test
+    steps:
+    - name: check if prod release
+      if: startsWith(github.ref, 'refs/tags/v')
+      run: |
+        echo "VERSION_TAG=${{ env.CRATER_VERSION }}"
+        echo "VERSION_TAG=${{ env.CRATER_VERSION }}" >> $GITHUB_ENV
+    - name: check if dev release
+      if: startsWith(github.ref, 'refs/tags/v') != true
+      run: |
+        echo "VERSION_TAG=${{ env.CRATER_VERSION }}-${{ github.run_number }}"
+        echo "VERSION_TAG=${{ env.CRATER_VERSION }}-${{ github.run_number }}" >> $GITHUB_ENV
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Login to DockerHub
+      uses: docker/login-action@v1 
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_TOKEN }}
+    - name: Docker Build and push crater-php
+      uses: docker/build-push-action@v2
+      with:
+        context: .
+        file: docker/Dockerfile
+        push: true
+        tags: craterapp/crater-php:${{env.VERSION_TAG}},craterapp/crater-php:latest
```

This would override latest tag with each push to master. Instead you could maybe push to `latest` for tags and to `next` for master builds.


 ```suggestion
        tags: craterapp/crater-php:${{env.VERSION_TAG}},craterapp/crater-php:${{ startsWith(github.ref, 'refs/tags/v') && 'next' ||  'latest' }}
```

---

Thanks @anbraten. So the workflow here is
- Every commit to master would create a docker image with tag `latest` and `<APP_VERSION>-<run-number>` e.g. `crater-php:7.0.0-23`. Here is the 7.0.0 is the next possible release candidate. `latest` always represent current code in master.
- When we cut a release from `master` and push a tag e.g. `7.0.0` - it would release an images with `<APP_VERSION>` i.e. basically the tag version `crater-php:7.0.0`

I have also refactored the code a bit to do better validations and make it more obvious. Please share your thoughts